src/utils.ts

```diff
@@ -157,32 +157,26 @@ export const updateIPDetails = async ({ ip, user, logger }): Promise<void> => {
     return
   }
 
-  let ipinfo
-
   try {
     // skip axios.get call if ip already exists in user object
     if (user.lastIPs.some((ipObject) => ipObject.ip === ip)) {
       return
     }
 
-    ipinfo = await fetchIP({ ip })
-  } catch (error) {
-    logger.info({ error }, "Failed to fetch ip details")
-  } finally {
-    try {
-      const res = await User.updateOne(
-        { "_id": user._id, "lastIPs.ip": ip },
-        { $set: { "lastIPs.$.lastConnection": Date.now() } },
+    const res = await User.updateOne(
+      { "_id": user._id, "lastIPs.ip": ip },
+      { $set: { "lastIPs.$.lastConnection": Date.now() } },
+    )
+
+    if (!res.nModified && yamlConfig.ipRecording.proxyChecking.enabled) {
+      const ipinfo = await fetchIP({ ip })
+      await User.findOneAndUpdate(
+        { "_id": user._id, "lastIPs.ip": { $ne: ip } },
```

Comment:
Is the `$ne` check still necessary if we already check the `res.nModified`?

Reply:
well, really not sure about the real usage of the api or parallel reqs, etc... but since there is no a global lock is better to have it